app/models/payout_report.rb

```diff
@@ -12,15 +14,36 @@ def encryption_key
     Rails.application.secrets[:attr_encrypted_key]
   end
 
-  def self.total_amount
-    PayoutReport.sum { |payout_report| payout_report.amount.to_i } / 1E18
+  def amount
+    PotentialPayment.where(payout_report_id: id).sum { |potential_payment| potential_payment.amount.to_i }
+  end
+
+  def fees
+    PotentialPayment.where(payout_report_id: id).sum { |potential_payment| potential_payment.fees.to_i }
   end
 
-  def self.total_fees
-    PayoutReport.sum { |payout_report| payout_report.fees.to_i } / 1E18
+  def num_payments
+    PotentialPayment.where(payout_report_id: id).count
   end
 
-  def self.total_payments
-    PayoutReport.sum(:num_payments)
+  # Updates the JSON summary of the report downloaded by admins
+  def update_report_contents
+    payout_report_json = JsonBuilders::PayoutReportJsonBuilder.new(payout_report: self).build
+    self.contents = payout_report_json.to_json
```

Comment:
IIRC you don't need to actually cast this to JSON format:

https://edgeguides.rubyonrails.org/active_record_postgresql.html#json-and-jsonb

Reply:
Yes I think that's right, but we do have other jsonb columns which are in json so I did this to keep it consistent.  I'm not opposed to not casting it to json though and move other jsonb columns to do the same in the future.

Reply:
Will keep as json for now.